test/test_helper.rb

```diff
@@ -128,7 +158,8 @@ def assert_logs_match_all(entry_list, in_order: false)
 
     def refute_logs_match(regexp)
       logging_assertion do |logs|
-        refute_match regexp, logs
```

Comment:
would [assert_contains](http://www.rubydoc.info/gems/minitest-rails-shoulda/0.4.1/MiniTest/Assertions:assert_contains) be better?

Reply:
That's in the `minitest-rails-shoulda` gem. The problem with `refute_match` is that it escapes the newline characters, so the logs in the error message are rather unreadable. You can add an additional error message, but then the logs are printed twice (which is a lot of text, half of which is useless).